### ISSUE TITLE
Contract compiled with ioctl produces empty bytecode

### ISSUE THREAD

**Issue by simonerom** [iotexproject/iotex-core#2581]
### What version of iotex-core are you using (`ioctl version`)?
```
Client:
gitStatus:"clean"  goVersion:"go version go1.15.5 darwin/amd64"  buildTime:"2020-12-01-GMT/17:06:39"

api.testnet.iotex.one:443:
packageVersion:"v1.1.3-30-g9ceb1109"  packageCommitID:"9ceb11098a1247a4d979ee735344cd5b6c6e8f03"  gitStatus:"clean"  goVersion:"go version go1.14.15 linux/amd64"  buildTime:"2021-03-05-UTC/06:01:19"
```

### What operating system and processor architecture are you using?
macOS Big Sur

### What did you do? If possible, provide a recipe for reproducing the error.
I'm attaching the solidity code, incl. abi and bytecode as generated by ioctl: [ownable.zip](https://github.com/iotexproject/iotex-core/files/6124049/ownable.zip)

```
$ ioctl contract compile Ownable ownable.sol --abi-out ownable.json --bin-out ownable.bin
$ ioctl contract deploy bin ownable.bin 
Error: failed to deploy contract with empty bytecode
```

When compiling in ide.iotex.io it also allows me to deploy it but of course no interaction is possible, as the data payload  empty

### What did you expect to see?
Some bytecode generated or an error during compilation

### What did you see instead?
Empty contract deployed from ide.iotex.io, empty bytecode and deployment error message from ioctl

**Comment by dustinxie** [comment]
asked uuz to check/repro in IDE

**Comment by Liuhaai** [comment]
Hi @simonerom, Could you check the binary results from solidity compiler? Here is the result of solidity compiler on my Laptop
`solc --bin ownable.sol`

> ======= ownable.sol:Ownable =======
Binary:

It seems that nothing is generated. I think the problem might be related to the solidity file not the ioctl tool

**Comment by simonerom** [comment]
@Liuhaai Yes you are right, I get an empty file also if I compile with `solc`.

I got this report from [Unify](ihttps://iotex.unifiprotocol.com) devs who could not interact with the Ownable contract after they deployed it to the IoTeX mainnet using ide.iotex.io. So I tried with ioctl and got this weird result.

Well, if it's not related to ioctl we can close this I guess 🤔
